deepchem/models/torch_models/lcnn.py

```diff
@@ -0,0 +1,428 @@
+               sitewise_n_feature: int = 25):
+    """
+        parameters
+        ----------
+
+        n_occupancy: int
+            number of possible occupancy
+        n_neighbor_sites_list: int
+            Number of neighbors of each site.
+        n_permutation: int
+            Diffrent permutations taken along diffrent directions.
+        dropout_rate: float
+            p value for dropout between 0.0 to 1.0
+        nconv: int
+            number of convolutions performed
+        n_feature: int
+            number of feature for each site
+        sitewise_n_feature: int
+            number of features for atoms for site-wise activation
+
+        """
+    super(LCNN, self).__init__()
+
+    modules = [LCNNBlock(n_occupancy * n_neighbor_sites, n_features)]
+    for i in range(n_conv - 1):
+      modules.append(
+          LCNNBlock(n_features * n_neighbor_sites, n_features, n_permutation))
+
+    self.LCNN_blocks = nn.Sequential(*modules)
+    self.Atom_wise_Conv = Atom_Wise_Convolution(n_features, sitewise_n_feature)
+    self.Atom_wise_Lin = Atom_Wise_Linear(sitewise_n_feature,
+                                          sitewise_n_feature)
+
+  def forward(self, G):
+    """
+        Parameters
+        ----------
+        G: DGLGraph
+            DGLGraph for a batch of graphs.
+
+        Returns
+        -------
+        y: torch.Tensor
+            A single scalar value
+
+        """
+    try:
+      import dgl
+    except:
+      raise ImportError("This class requires DGL to be installed.")
+    node_feats = G.ndata.pop('x')
+
+    for conv in self.LCNN_blocks:
+      node_feats = conv(G, node_feats)
+    node_feats = self.Atom_wise_Conv(node_feats)
+    node_feats = self.Atom_wise_Lin(node_feats).sum(axis=1)
+    G.ndata['new'] = node_feats
+
+    y = dgl.mean_nodes(G, 'new')
+    return y
+
+
+class LCNNModel(TorchModel):
+  """Lattice Convolutional Neural Network (LCNN).
+    Here is a simple example of code that uses the LCNNModel with
+    Platinum 2d Adsorption dataset.
+
+    - import deepchem as dc
+    - tasks, datasets, transformers = dc.molnet.load_Platinum_Adsorption()
+    - train, valid, test = datasets
+    - model = dc.models.LCNNModel(mode='regression', batch_size=8, learning_rate=0.001)
+    - model.fit(train, nb_epoch=10)
+
+    This model takes arbitrary configurations of Molecules on an adsorbate and predicts
+    their formation energy. These formation energies are found using DFT calculations and
+    LCNNModel is to automate that process. This model defines a crystal graph using the
+    distance between atoms. The crystal graph is an undirected regular graph (equal neighbours)
+    and different permutations of the neighbours are pre-computed using the LCNNFeaturizer.
+    On each node for each permutation, the neighbour nodes are concatenated which are further operated.
+    This model has only a node representation. Please confirm the detail algorithms from [1]_.
+
```

Comment:
Can you add a usage example to this docstring?

Reply:
That can be added only after the LCNN featurisation gets merged otherwise, it might a give doctest error

Reply:
Ah good point. Ok will resolve this issue